packages/jspsych/src/ProgressBar.ts

```diff
@@ -31,6 +35,16 @@ export class ProgressBar {
   /** Updates the progress bar according to `this.progress` */
   private update() {
     this.innerDiv.style.width = this._progress * 100 + "%";
+
+    this.updateMessage();
+  }
+
+  private updateMessage() {
```

Comment:
@Shaobin-Jiang I missed that we always need this together with `update()` anyway – would you mind just joining it into the existing `update` function? We're already calling update in `setupElements()` too... Thanks for all the changes and the unit test – looks very good!

Reply:
Done.